src/qt/bitcoingui.cpp

```diff
@@ -664,14 +664,15 @@ void BitcoinGUI::addWallet(WalletModel* walletModel)
 {
     if (!walletFrame) return;
     if (!walletFrame->addWallet(walletModel)) return;
-    const QString display_name = walletModel->getDisplayName();
-    setWalletActionsEnabled(true);
     rpcConsole->addWallet(walletModel);
-    m_wallet_selector->addItem(display_name, QVariant::fromValue(walletModel));
-    if (m_wallet_selector->count() == 2) {
+    if (m_wallet_selector->count() == 0) {
+        setWalletActionsEnabled(true);
+    } else if (m_wallet_selector->count() == 1) {
```

Comment:
Why does this change from 2 to 1?

Reply:
Because `m_wallet_selector->addItem()` is called after this code now.

Reply:
Why not just add the `count == 1` case in the old code?

Reply:
> Why not just add the `count == 1` case in the old code?

It would change wallet selector behavior, and it won't fix a bug due to the unconditional `setWalletActionsEnabled(true)` call.

> Edit: if it's due to some other signal/slot then I'd say to refactor to remove that dependency.

Not sure what do you mean. FWIW, moving this logic into `setCurrentWalletBySelectorIndex()` or `setCurrentWallet()` won't work as these methods do not know if a wallet has been added or removed.